x-pack/plugins/upgrade_assistant/server/routes/deprecation_logging.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import {
+  getDeprecationLoggingStatus,
+  setDeprecationLogging,
+} from '../lib/es_deprecation_logging_apis';
+import { versionCheckHandlerWrapper } from '../lib/es_version_precheck';
+import { RouteDependencies } from '../types';
+
+export function registerDeprecationLoggingRoutes({ router }: RouteDependencies) {
+  router.get(
+    {
+      path: '/api/upgrade_assistant/deprecation_logging',
+      validate: false,
+    },
+    versionCheckHandlerWrapper(
```

Comment:
Don't all the routes in upgrade assistant also require license check?

Reply:
🤔that is a good question. We should follow up on this one (it was not there in the original logic).